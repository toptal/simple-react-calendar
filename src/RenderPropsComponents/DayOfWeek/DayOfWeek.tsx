import React, { FC } from 'react'
import classnames from 'classnames'

export type Props = {
  blockClassName?: string
  day: string
  isWeekend: boolean
  // TODO: fix type definitions instead
  key?: string
}

const DayOfWeek: FC<Props> = ({ blockClassName, isWeekend, day }) => (
  <div
    className={classnames(`${blockClassName}-days_of_week_day`, {
      'is-weekend': isWeekend
    })}
  >
    {day}
  </div>
)

export default DayOfWeek
