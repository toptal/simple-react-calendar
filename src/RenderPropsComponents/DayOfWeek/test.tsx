import { render } from '@testing-library/react'
import React, { ComponentProps } from 'react'

import DayOfWeek from './DayOfWeek'

const renderDayOfWeek = (
  props: Partial<ComponentProps<typeof DayOfWeek>> = {}
) => {
  const baseProps = {
    blockClassName: 'example-class-name',
    day: '2015-12-12',
    isWeekend: true,
    ...props
  }

  // eslint-disable-next-line react/jsx-props-no-spreading
  return render(<DayOfWeek {...baseProps} />)
}

test('default render', () => {
  const { container } = renderDayOfWeek()

  expect(container.firstChild).toMatchSnapshot()
})
