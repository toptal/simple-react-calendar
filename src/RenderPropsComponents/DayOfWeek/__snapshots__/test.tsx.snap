// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render 1`] = `
<div
  class="example-class-name-days_of_week_day is-weekend"
>
  2015-12-12
</div>
`;
